test/androidTest/java/org/thoughtcrime/securesms/espressoTests/ConversationDeleteTest.java

```diff
@@ -0,0 +1,50 @@
+package org.thoughtcrime.securesms.espressoTests;
+
+import android.support.test.rule.ActivityTestRule;
+import android.support.test.runner.AndroidJUnit4;
+import android.test.suitebuilder.annotation.LargeTest;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.thoughtcrime.securesms.ConversationListActivity;
+import org.thoughtcrime.securesms.R;
+import org.thoughtcrime.securesms.util.ConversationActions;
+import org.thoughtcrime.securesms.util.Expectations;
+
+@LargeTest
+@RunWith(AndroidJUnit4.class)
+public class ConversationDeleteTest {
+
+    @Rule
+    public ActivityTestRule<ConversationListActivity> mActivityTestRule = new ActivityTestRule<>(ConversationListActivity.class);
+
+    @Test
+    public void deleteConversationSimpleTest() throws InterruptedException {
+
+        String convoRecipient = "+1119991111";
+
+        ConversationActions.createNewConversationWithImmediateExit(convoRecipient, "hello");
+        Expectations.checkIsDisplayed(R.id.from, convoRecipient);
+
+        ConversationActions.deleteConversation(convoRecipient);
+        Thread.sleep(6000);
+        Expectations.checkDoesNotExist(R.id.from, convoRecipient);
+
+    }
+
+    @Test
+    public void deleteConversationAndUndoTest() throws InterruptedException {
+
+        String convoRecipient = "+1119991112";
+
+        ConversationActions.createNewConversationWithImmediateExit(convoRecipient, "hello");
+        Expectations.checkIsDisplayed(R.id.from, convoRecipient);
+
+        ConversationActions.deleteConversation(convoRecipient);
+        Thread.sleep(1000);
+        ConversationActions.undoDeletion();
```

Comment:
Maybe add some sleep after this, cause even if this action didn't do anything I think the test would still pass.

Reply:
actually I might add a check after the delete to make sure it doesn't exist. then the check will be 100% guaranteed.